ogusa_calibrate/estimate_beta_j.py

```diff
@@ -1,94 +1,145 @@
 
-
-def beta_estimate(beta_initial_guesses, og_spec={}, two_step=False,
-                  client=None):
-    '''
+# Define default betas
+A = 0.96
+E = 0.005
+BETAS_DEFAULT = [
+    A - 4 * E,
+    A - 2 * E,
+    A - E,
+    A,
+    min(A + E, 0.99),
+    min(A + 2 * E, 0.99),
+    min(A + 4 * E, 0.99),
+]
+
+
+def beta_estimate(
+    betas_initial_guess=BETAS_DEFAULT,
+    history_path="history_path.pickle",
+    og_spec={},
+    two_step=False,
+    client=None,
+    overwrite=False,
+):
+    """
     This function estimates the beta_j parameters using a simulated
     method of moments estimator that targets moments of the wealth
     distribution.
 
     Args:
-    beta_initial_guesses (array-like): array of initial guesses for the
-        beta_j parameters
+    betas_initial_guess (array-like): array of initial guesses for the
+        beta_j parameters. Passing none passes BETAS_DEFAULT.
     og_spec (dict): any updates to default model parameters
     two_step (boolean): whether to use a two-step estimator
     client (Dask Client object): dask client for multiprocessing
+    overwrite: does not overwrite history_path by default
 
     Returns:
         beta_hat (array-like): estimates of the beta_j
         beta_se (array-like): standard errors on the beta_j estimates
 
-    '''
+    """
+
+    # If there is already a history of iterations,
+    # set the initial guess to betas_initial_guess
+    if os.path.isfile(history_path):
+        history = pd.read_pickle(history_path).sort_values("distance")
+        betas_initial_guess = history.iloc[0, :7].tolist()
+    else:
+        history = pd.DataFrame()
+
+    new_history = []
+    new_history = pd.DataFrame(new_history)
 
     # initialize parametes object
     tax_func_path = os.path.join(
-        '..', '..', 'dynamic', 'ogusa', 'data', 'tax_functions',
-        'TxFuncEst_baseline_PUF.pkl')
-    p = Specifications(baseline=True)
+        "ogusa_calibrate",
+        "data",  # add ogusa above for last ditch effort
+        "TxFuncEst_baseline_PUF.pkl",
+    )
+    print(tax_func_path)
+    print(os.path.isfile(tax_func_path))
+    p = Specifications(baseline=True, tax_func_path=tax_func_path)
     p.update_specifications(og_spec)
     p.get_tax_function_parameters(client, False, tax_func_path)
 
     # Compute wealth moments from the data
-    scf = wealth.get_wealth_data(
-            scf_yrs_list=[2019], web=True, directory=None)
-    data_moments = wealth.compute_wealth_moments(
-            scf, p.lambdas)
+    scf = wealth.get_wealth_data(scf_yrs_list=[2019], web=True, directory=None)
+    data_moments = wealth.compute_wealth_moments(scf, p.lambdas)
 
     # Get weighting matrix
     W = compute_weighting_matrix(p, optimal_weight=False)
 
     # call minimizer
     # set bounds on beta estimates (need to be between 0 and 1)
-    bnds = np.tile(np.array([1e-12, 1]), (p.J, 1))  # Need (1e-12, 1) J times
+    bnds = np.tile(
+        np.array([0.85, 0.995]), (p.J, 1)
```

Comment:
@jwal1024 Your issue regarding the size of the bounds is here.  Since you are now estimating only two parameters (the mean `beta` and `E`) you need a bounds vector of length 2.